src/config.rs

```diff
@@ -9,6 +9,19 @@ use std::io::Read;
 use std::io::Write;
 use std::path::PathBuf;
 
+#[derive(Clone, Deserialize)]
+pub enum SASLMethod {
+    // Only PLAIN SASL is supported for now.
+    PLAIN
+}
+
+#[derive(Clone, Deserialize)]
+pub struct Auth {
```

Comment:
Please remove `sasl_method` field (with `SASLMethod` type) and rename this type as `SASLAuth`. We currently support multiple auth methods implicitly, like the `pass` field and `auto_cmds` (which can be used to authenticate via `NickServ`), so this is not a general `Auth` type, it's a type specific to `SASL` method of authentication.

Reply:
renamed.